File: examples/haskell-functional/tuple.hs
module Tuple where {
identity value = value;

choose :: Bool -> (Int, Int) -> Int;
choose flag pair =
    if flag
        then case pair of { (left, right) -> left + right }
        else 0;

gpuMain =
    let {
        pair = identity (20, 22);
        flag = identity True
    } in choose flag pair
}

File: examples/rust-functional/tuple.rs
fn choose(flag: bool, pair: (i32, i32)) -> i32 {
    if flag {
        match pair {
            (left, right) => left + right,
        }
    } else {
        0
    }
}

fn gpu_main() -> i32 {
    let pair = (20, 22);
    choose(true, pair)
}

fn main() {}
